integration/tests/epochs/recover_epoch/suite.go

```diff
@@ -12,10 +24,85 @@ type Suite struct {
 func (s *Suite) SetupTest() {
 	// use a shorter staking auction because we don't have staking operations in this case
 	s.StakingAuctionLen = 2
-	s.DKGPhaseLen = 50
-	s.EpochLen = 250
+	s.DKGPhaseLen = 10
+	s.EpochLen = 80
 	s.EpochCommitSafetyThreshold = 20
 
 	// run the generic setup, which starts up the network
 	s.BaseSuite.SetupTest()
 }
+
+// getNodeInfoDirs returns the internal node private info dir and the node config dir from a container with the specified role.
+func (s *Suite) getNodeInfoDirs(role flow.Role) (string, string) {
+	internalNodePrivInfoDir := fmt.Sprintf("%s/%s", s.GetContainersByRole(role)[0].BootstrapPath(), bootstrap.DirPrivateRoot)
+	nodeConfigJson := fmt.Sprintf("%s/%s", s.GetContainersByRole(role)[0].BootstrapPath(), bootstrap.PathNodeInfosPub)
+	return internalNodePrivInfoDir, nodeConfigJson
+}
+
+// executeEFMRecoverTXArgsCMD executes the efm-recover-tx-args CLI command to generate EpochRecover transaction arguments.
+// Args:
+//
+//	role: the container role that will be used to read internal node private info and the node config json.
+//	snapshot: the protocol state snapshot.
+//	collectionClusters: the number of collector clusters.
+//	numViewsInEpoch: the number of views in the recovery epoch.
+//	numViewsInStakingAuction: the number of views in the staking auction of the recovery epoch.
+//	epochCounter: the epoch counter.
+//	targetDuration: the target duration for the recover epoch.
+//	targetEndTime: the target end time for the recover epoch.
+//	randomSource: the random source of the recover epoch.
+//	out: the tx args output file full path.
+func (s *Suite) executeEFMRecoverTXArgsCMD(collectionClusters, numViewsInEpoch, numViewsInStakingAuction, epochCounter, targetDuration, targetEndTime uint64, out string) {
+	// read internal node info from one of the consensus nodes
+	internalNodePrivInfoDir, nodeConfigJson := s.getNodeInfoDirs(flow.RoleConsensus)
+	an1 := s.GetContainersByRole(flow.RoleAccess)[0]
+	anAddress := an1.Addr(testnet.GRPCPort)
+	// set command line arguments
+	os.Args = []string{
```

Comment:
The portion of bootstrapping logic which is shared between the Bootstrapping CLI and the integration tests is factored out into the shared `run` package, which is used by both. For example, here is the shared function to generate a root QC for a new network: 
https://github.com/onflow/flow-go/blob/f97edce229aa86643125a6f7c78af8e43a7d5e04/cmd/bootstrap/run/qc.go#L61

If it isn't too much additional work, it would be nice to do the same here to avoid the more cumbersome interaction via shell-style interface and flags.

Reply:
https://github.com/onflow/flow-go/pull/6156/commits/3b6d7078caa1f0e14b4e3ea13c372366496c9fdb